CMakeLists.txt

```diff
@@ -21,50 +18,51 @@ endif()
 #####################################
 # Define Options
 #####################################
-option(BUILD_EXAMPLES "Builds the examples in the project" ON)
-option(BUILD_TESTING  "Builds the tests in the project"    ON)
+option(CROW_BUILD_EXAMPLES "Build the examples in the project" ON )
+option(CROW_BUILD_TESTS    "Build the tests in the project"    ON )
+option(CROW_AMALGAMATE     "Combine all headers into one"      OFF)
 
 #####################################
-# Define CMake Module Imports
+# Define Targets
 #####################################
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/dependencies.cmake)
-include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/compiler_options.cmake)
+add_library(crow INTERFACE)
+target_include_directories(crow INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
-#####################################
-# Define project-wide imports
-#####################################
-# this can be alternatively (and as recommended way) done with target_include_directories()
-if(BUILD_EXAMPLES OR BUILD_TESTING)
-	set(PROJECT_INCLUDE_DIR
-		${CMAKE_CURRENT_SOURCE_DIR}/include
-	)
+find_package(Boost COMPONENTS date_time REQUIRED)
```

Comment:
what about the version `1.64.0` and the components `system` and `thread` that were in the old dependencies file?

Reply:
Is version `1.64` mandatory?
The `vcpkg.json` file is pulling in version `1.70.0` of Boost for the examples and it's working just fine.

Since `1.69` `Boost.System` is header-only and doesn't need to be linked anymore.

And maybe I've missed something, but I couldn't find a use of `Boost.Thread`.

I've added the version constraint back in. 👍

Reply:
`1.64` is there for backwards compatibility with older OSes, It was set by the Conan team when they were adding Crow to conan-center. I think it would be best to keep the minimum version of boost that Crow can actually run under

Reply:
[Conan-Center](https://conan.io/center/crowcpp-crow?tab=dependencies) is reporting a `boost/1.75.0` dependency. Am I missing something?

Reply:
True, like I said they set it up that way (see https://github.com/CrowCpp/Crow/pull/96#issuecomment-758626540)

Reply:
No, it isn't. I'll take care of it. 👍